### PR TITLE
Stop incrementing `jit_entry_calls` once threshold is hit

### PR DESCRIPTION
**Pull request by byroot**
Otherwise the ISeq page will constantly be written into preventing it from being shared.

cc @k0kubun as we mentioned that recently.

### CODE REVIEW

**Comment by byroot** [comment]
Hum, RJIT tests are failing in a weird way, not sure why.

**Comment by k0kubun** [comment]
You're returning `body->jit_entry` even when `rb_rjit_call_p` is false. `rb_rjit_call_p` means "call RJIT-compiled code", not "compile code with RJIT". When it's false, it must return `NULL` for invalidation reasons.

Also, with today's master, you'd want to change `jit_compile_exception` and `body->jit_exception_calls` as well.

**Comment by byroot** [comment]
> When it's false, it must return `NULL` for invalidation reasons.

Ah thank you, I totally missed that.

**Review by k0kubun — APPROVED** [review]
:+1:

File: vm.c [hunk 1]
```diff
@@ -439,16 +436,16 @@ jit_compile_exception(rb_execution_context_t *ec)
     // Increment the ISEQ's call counter
```
Comment (k0kubun) [inline vm.c:436]:
```suggestion
```

File: vm.c [hunk 2]
```diff
@@ -381,19 +381,16 @@ static VALUE vm_invoke_proc(rb_execution_context_t *ec, rb_proc_t *proc, VALUE s
...
-    if (yjit_enabled || rb_rjit_call_p) {
-        body->jit_entry_calls++;
-    }
-    else {
+    if (!(yjit_enabled || rb_rjit_call_p)) {
         return NULL;
     }
 
-    // Trigger JIT compilation if not compiled
+    // Increment the ISEQ's call counter or Trigger JIT compilation if not compiled
```
Comment (k0kubun) [inline vm.c:391]:
```suggestion
    // Increment the ISEQ's call counter and trigger JIT compilation if not compiled
```

File: vm.c [hunk 3]
```diff
@@ -439,16 +436,16 @@ jit_compile_exception(rb_execution_context_t *ec)
...
     struct rb_iseq_constant_body *body = ISEQ_BODY(iseq);
-    if (rb_yjit_compile_new_iseqs()) {
-        body->jit_exception_calls++;
-    }
-    else {
+    if (!rb_yjit_compile_new_iseqs()) {
         return NULL;
     }
 
     // Trigger JIT compilation if not compiled
```
Comment (k0kubun) [inline vm.c:443]:
```suggestion
    // Increment the ISEQ's call counter and trigger JIT compilation if not compiled
```

**Pull request merged**